### PR TITLE
[enhance] automatic batch creation, move and split

### PR DESCRIPTION
**Pull request by rmehta**
Pending

- [x] Automatic batch selection for outgoing (Stock Entry, Delivery Note)

### Setting in Item to automatically create new Batches

item_setup_for_batch

### Batch Move

batch-move

### Batch Split

batch-split

### CODE REVIEW

**Comment by jof2jc** [comment]
@rmehta, what about new incoming batches on purchase receipt/invoice and stock entry? Will the system auto-create it like serial no?

**Comment by nabinhait** [comment]
@rmehta Should I check and test this PR now? Or I should wait for your pending task.

File: erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json [hunk 1]
```diff
@@ -1040,11 +1040,11 @@
    "unique": 0
   }, 
   {
-   "allow_on_submit": 0, 
+   "allow_on_submit": 1, 
    "bold": 0, 
    "collapsible": 0, 
    "columns": 0, 
-   "fieldname": "allow_zero_valuation_rate", 
+   "fieldname": "is_sample_item",
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json]: "is_sample_item" has been renamed to "allow_zero_valuation_rate" last week.

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json (hunk 1 above)
Comment (rmehta) [inline erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json]: done! Good catch

**Pull request merged**